File: go/src/leetcode/offer/Offer6.go
package offer

/**
 * @author wujingxinit@outlook.com
 * @date 2022/12/28 14:58
 */

func reversePrint(head *ListNode) []int {
	list := make([]int, 0)
	var dfs func(head *ListNode)
	dfs = func(head *ListNode) {
		if head == nil {
			return
		}
		dfs(head.Next)
		list = append(list, head.Val)
	}
	dfs(head)
	return list
}

File: java/src/leetcode/offer/Offer6.java
package leetcode.offer;

import common.*;

import java.util.*;

/**
 * @author wujingxinit@outlook.com
 * @date 2023/9/12 13:26
 */
public class Offer6 {

    public int[] reversePrint(ListNode head) {
        List<Integer> list = new ArrayList<>();
        dfs(head, list);
        int[] res = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            res[i] = list.get(i);
        }
        return res;
    }

    private void dfs(ListNode head, List<Integer> list) {
        if (head == null) {
            return;
        }
        dfs(head.next, list);
        list.add(head.val);
    }
}
